### PR TITLE
Switch to Github Actions on master

### PR DESCRIPTION
**Pull request by haonanya**
For in-tree build, opencl-clang depends on llvm-project which is
frequently updated everyday, so it is not necessary to have patch
mechanism for clang and spirv which will result in many errors when apply patches.
Currently remove the patch only on master branch.

Signed-off-by: haonanya <haonan.yang@intel.com>

### CODE REVIEW

**Comment by haonanya** [comment]
Remove-__IMAGE_SUPPORT__-macro-for-SPIR.patch is to fix OpenCL CTs test_conformance/basic/test_preprocessors.cpp which failed on some device without __IMAGE_SUPPORT__ support. I am not sure if the patch is right fix to undefine __IMAGE_SUPPORT__ for SPIR target directly.

File: options.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ OpenCLArgList *OpenCLOptTable::ParseArgs(const char *szOptions,
     }
 
     unsigned prev = index;
-    Arg *pArg = ParseOneArg(*pArgs, index);
+    Arg *pArg = ParseOneArg(*pArgs, index).release();
```
Comment (hewj03) [inline options.cpp:92]: will "Arg *pArg" be released later?

Comment (haonanya) [reply options.cpp:92]: It is because API change from 
`Arg *ParseOneArg(const ArgList &Args, unsigned &Index,
                   unsigned FlagsToInclude = 0,
                   unsigned FlagsToExclude = 0) const;` 
to
 `std::unique_ptr<Arg> ParseOneArg(const ArgList &Args, unsigned &Index,
                                   unsigned FlagsToInclude = 0,
                                   unsigned FlagsToExclude = 0) const;`
There are compiling errors: cannot convert ‘std::unique_ptr<llvm::opt::Arg>’ to ‘llvm::opt::Arg*’ in initialization.

Comment (hewj03) [reply options.cpp:92]: please check whether pArg is released later.

Comment (haonanya) [reply options.cpp]: Update patch, please review it, thanks very much.

File: options_compile.cpp [hunk 2]
```diff
@@ -178,7 +178,6 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
   effectiveArgs.push_back("-cl-kernel-arg-info");
   effectiveArgs.push_back("-fno-validate-pch");
   effectiveArgs.push_back("-fno-caret-diagnostics");
-  effectiveArgs.push_back("-dwarf-column-info");
```
Comment (hewj03) [inline options_compile.cpp:181]: why this change? is this outdate code?

Comment (haonanya) [reply options_compile.cpp:181]: Yes, with community commit https://reviews.llvm.org/rGb0b5162fc23c55906a461366f8acef2431d951c5
option -dwarf-column-info is no longer exist.

Comment (hewj03) [reply options_compile.cpp:181]: I see, thanks

**Review by hewj03 — APPROVED** [review]
